tools/buildsys/src/manifest.rs

```diff
@@ -405,34 +476,44 @@ impl SupportedArch {
 }
 
 #[derive(Deserialize, Debug, PartialEq, Eq, Hash)]
-#[serde(rename_all = "kebab-case")]
-pub(crate) enum GrubFeature {
-    SetPrivateVar,
+#[serde(try_from = "String")]
+pub enum ImageFeature {
+    GrubSetPrivateVar,
+}
+
+impl TryFrom<String> for ImageFeature {
+    type Error = Error;
+    fn try_from(s: String) -> Result<Self> {
+        match s.as_str() {
+            "grub-set-private-var" => Ok(ImageFeature::GrubSetPrivateVar),
```

Comment:
If interested, serde plain can do this for you. [serde_plain::derive_fromstr_from_deserialize](https://docs.rs/serde_plain/latest/serde_plain/macro.derive_fromstr_from_deserialize.html)

Reply:
Perhaps I'm not clear how to use it, but it seems to lead to relatively more boilerplate.

Without the `#[serde(try_from = "String")]` container attribute, deserialization fails with an error like this, because TOML keys are always strings:
```
invalid type: string "grub-set-private-var", expected enum ImageFeature for key ...
```

With the container attribute, the type is first deserialized into a string and then converted to the enum.

`derive_fromstr_from_deserialize` gives me a `FromStr` method, but I don't see a way to get `serde` to use that method without implementing a custom deserialize method.

(I also turned up `serde_with::DeserializeFromStr`, but we don't use `serde_with` in the project today and I didn't feel like it was worth bringing in a new dependency.)

Reply:
not worth the hassle i guess, sorry for sending you down that road.